// erela/dist/src/entities/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersEntity = exports.connectWithDB = void 0;
const typeorm_1 = require("typeorm");
const dotenv_1 = require("dotenv");
const path_1 = require("path");
require("reflect-metadata");
require("colors");
dotenv_1.config();
const { DB_HOST, DB_USER, DB_PASS, TYPEORM_DATABASE, EXT } = process.env;
exports.connectWithDB = async () => {
    const connection = await typeorm_1.createConnection({
        type: 'sqlite',
        database: TYPEORM_DATABASE,
        entities: [`/**/*.${EXT}`].map((path) => path_1.join(__dirname, path)),
        subscribers: [`/**/*.${EXT}`].map((path) => path_1.join(__dirname, '../subscribers', path)),
    });
    await connection
        .synchronize(false) // ! pass true to drop everything and create them again [ DO NOT DO IT ON PROD ]
        // tslint:disable-next-line: no-console
        .then(() => console.log('💾 Connected with DB 💾'.green.bold))
        .catch(() => console.error('Faild to sync with DB'.red.bold));
    return connection;
};
var users_entity_1 = require("./users.entity");
Object.defineProperty(exports, "UsersEntity", { enumerable: true, get: function () { return users_entity_1.UsersEntity; } });
